pandas/io/sql.py

```diff
@@ -692,37 +692,26 @@ def insert_data(self):
         column_names = list(map(str, temp.columns))
         ncols = len(column_names)
         data_list = [None] * ncols
-        blocks = temp._data.blocks
-
-        for b in blocks:
-            if b.is_datetime:
-                # return datetime.datetime objects
-                if b.is_datetimetz:
-                    # GH 9086: Ensure we return datetimes with timezone info
-                    # Need to return 2-D data; DatetimeIndex is 1D
-                    d = b.values.to_pydatetime()
-                    d = np.atleast_2d(d)
-                else:
-                    # convert to microsecond resolution for datetime.datetime
-                    d = b.values.astype("M8[us]").astype(object)
-            elif b.is_timedelta:
-                # numpy converts this to an object array of integers,
-                #  whereas b.astype(object).values would convert to
-                #  object array of Timedeltas
-                d = b.values.astype(object)
+
+        for i in range(len(temp.columns)):
```

Comment:
Is the issue with using `.items` that it can provide a frame back in case of duplicate labels? If so wonder if we shouldn't just update `.items` to do what you have done here

Reply:
you mean BlockManager.items?  i dont see how thats relevant here

Reply:
`temp` is a DataFrame, no?

Reply:
yes.  this is just doing this column-wise instead of block-wise to avoid accessing `._data`